api/python/provisioner/commands/setup_provisioner.py

```diff
@@ -1823,6 +1827,12 @@ def _run(self, nodes, **kwargs):  # noqa: C901 FIXME
                 targets=ALL_MINIONS
             )
 
+        logger.info("Refresh enclosure id on the system")
+        ssh_client.cmd_run(
+            "salt-call state.apply components.provisioner.config.enclosure_id",
```

Comment:
we need to run "salt-call state.apply components.provisioner.config.enclosure_id.set_pillar" as separate state otherwise it will set old enclosure id again in pillar.

1.  salt-call state.apply components.provisioner.config.enclosure_id
2.  salt-call state.apply components.provisioner.config.enclosure_id.set_pillar

Reply:
done